projects/ipykernel/fuzz_unpack_roundtrip.py

```diff
@@ -18,29 +18,35 @@
 from ipykernel.serialize import unpack_apply_message, pack_apply_message
 
 
+def ConsumeRandomLengthBufferList(fdp):
+    """Creates a list of buffers of various lengths"""
+    buffers = []
+    max_range = fdp.ConsumeIntInRange(3, 50)
+    for _ in range(3, max_range):
+        buffers.append(fdp.ConsumeBytes(fdp.ConsumeIntInRange(0, 200)))
+    return buffers
+
+
 def TestOneInput(data):
-  if len(data) < 48:
-    return
-  fdp = atheris.FuzzedDataProvider(data)
-  buffers = [
-    fdp.ConsumeBytes(1048),
-    fdp.ConsumeBytes(1048),
-    fdp.ConsumeBytes(1048),
-  ]
-  try:
-    f, args, kwargs = unpack_apply_message(buffers)
-  except Exception as e:
-    return
-
-  # Anything unpackable, should be packable.
-  pack_apply_message(f, args, kwargs)
+    if len(data) < 48:
+        return
+    fdp = atheris.FuzzedDataProvider(data)
+
+    try:
+        buffers = ConsumeRandomLengthBufferList(fdp)
+        f, args, kwargs = unpack_apply_message(buffers)
+    except Exception:
+        return
+
+    # Anything unpackable, should be packable.
+    pack_apply_message(f, args, kwargs)
 
 
 def main():
-  atheris.instrument_all()
-  atheris.Setup(sys.argv, TestOneInput)
-  atheris.Fuzz()
+    atheris.instrument_all()
+    atheris.Setup(sys.argv, TestOneInput)
+    atheris.Fuzz()
```

Comment:
Could we keep the two spaces as indentation in this file?

Reply:
Done